### PR TITLE
Nexus: fix convert4qmc hdf5 issue

### PR DESCRIPTION
**Pull request by jtkrogel**
## Proposed changes

Fix an issue with expected filenames.  `convert4qmc` no longer produces an hdf5 file when given one from PySCF.  This leads to incorrect paths being provided to QMCPACK for the orbitals.  Fix this by inserting a symlink in place of the file convert4qmc used to generate.

## What type(s) of changes does this code introduce?

- Bugfix

### Does this introduce a breaking change?

- No

## What systems has this change been tested on?

Laptop

## Checklist

- Yes. This PR is up to date with current the current state of 'develop'

### CODE REVIEW

**Review by prckent — COMMENTED** [review]
Can you please put a comment about why this is done?

**Comment by jtkrogel** [comment]
Done.

**Review by prckent — APPROVED** [review]
LGTM

File: nexus/lib/qmcpack_converters.py [hunk 1]
```diff
@@ -864,6 +864,19 @@ def check_sim_status(self):
         output = open(os.path.join(self.locdir,self.outfile),'r').read()
         #errors = open(os.path.join(self.locdir,self.errfile),'r').read()
 
+        # Recent versions of convert4qmc no longer produce the orbs.h5 file.
```
Comment (ye-luo) [inline nexus/lib/qmcpack_converters.py:867]: Do you know how recent? If not, put 3.15 should be fine.

**Comment by markdewing** [comment]
Where does .orbs.h5 get used?  Is it possible to propagate the name of the h5 file where it is needed, rather than make symbolic link?

**Comment by markdewing** [comment]
It looks like there are cases in convert4qmc that can still produce an `.orbs.h5` file (-dirac and -hdf5 flags).  Will cause this code to try make symlink to the same file? (It'll probably issue a failure message for `ln`, but still continue.)

File: nexus/lib/qmcpack_converters.py (hunk 1 above)
Comment (jtkrogel) [reply nexus/lib/qmcpack_converters.py:867]: It's not very recent.  @anbenali knows better.

**Comment by jtkrogel** [comment]
Yes, it would try and fail to make a symlink in those cases but proceed fine anyway.  I can put in a file existence check before proceeding with the attempted symlink creation.

**Comment by jtkrogel** [comment]
More safety checks added.  Now the symlink will only be made if an h5 file has been provided and one has not been created by convert4qmc.

File: nexus/lib/qmcpack_converters.py (hunk 1 above)
Comment (anbenali) [reply nexus/lib/qmcpack_converters.py:867]: convert4qmc produces an HDF5 only when a txt file is provided (gamess) and the tag -hdf5 is produced.

**Review by anbenali — APPROVED** [review]

**Comment by prckent** [comment]
Test this please

**Pull request merged**